pulsar-storm/src/main/java/org/apache/pulsar/storm/PulsarSpout.java

```diff
@@ -215,6 +204,36 @@ public void emitNextAvailableTuple() {
         }
     }
 
+    private boolean emitFailedMessage() {
+        Message<byte[]> msg;
+
+        while ((msg = failedMessages.peek()) != null) {
+            MessageRetries messageRetries = pendingMessageRetries.get(msg.getMessageId());
+            if (messageRetries != null) {
```

Comment:
Move this if statement to a separate function.

Reply:
I wouldn't much worry about it because this piece of code is not being reused and `nextTuple` anyway is calling `emitNextAvailableTuple` and I don't think we want to do another method call for this small logic.